File: 10000~/10828.py
# 스택

import sys

n = int(sys.stdin.readline())

com = str()
x = int()
arr = list()

def push():
    arr.append(x)

def pop():
    if len(arr) > 0:
        print(arr[len(arr) - 1])
        del arr[len(arr) - 1]
    else:
        print(-1)

def size():
    print(len(arr))

def empty():
    if len(arr) == 0:
        print(1)
    else:
        print(0)

def top():
    if len(arr) > 0:
        print(arr[len(arr) - 1])
    else:
        print(-1)

cmd = {
    "push": push,
    "pop": pop,
    "size": size,
    "empty": empty,
    "top": top
}

for i in range(n):
    com = sys.stdin.readline().strip()
    
    for i in com:
        if i == ' ':
            com, x = com.split()
            break

    cmd[com]()

File: 10000~/10828.cpp
// 스택

#include <iostream>

int dat[10000];
int pos = 0;

void push(int x);
int pop();
int size();
bool empty();
int top();

int main() {
	std::ios::sync_with_stdio(0);
	std::cin.tie(0);

	int n;
	std::cin >> n;

	while (n--) {
		std::string cmd;
		std::cin >> cmd;

		if (cmd == "push") {
			int x;
			std::cin >> x;
			push(x);
		}
		else if (cmd == "pop") std::cout << pop() << '\n';
		else if (cmd == "size") std::cout << size() << '\n';
		else if (cmd == "empty") std::cout << empty() << '\n';
		else if (cmd == "top") std::cout << top() << '\n';
	}	

	return 0;
}

void push(int x) {
	dat[pos++] = x;
}

int pop() {
	if (empty()) return -1;
	else return dat[--pos];
}

int size() {
	return pos;
}

bool empty() {
	if (size()) return 0;
	else return 1;
}

int top() {
	if (empty()) return -1;
	else return dat[pos - 1];
}